### PR TITLE
refactor(vendor): tiny not important refactors

### PR DESCRIPTION
**Pull request by weihanglo**
### What does this PR try to resolve?

Did some tiny refactors in cargo vendor I found when working on other stuff.

### How should we test and review this PR?

~~I believe `std::path::Path` should be quite cross-platform compatible, as [it calls into something like [`GetFullPathNameW`](
https://github.com/rust-lang/rust/blob/e760daa6a729b3d52a38804e9766f7d89dc27357/library/std/src/sys/path/windows.rs#L280-L287) eventually when being used. I might be wrong though.~~

See https://github.com/rust-lang/cargo/pull/13610#discussion_r1532741537

### CODE REVIEW

File: src/cargo/ops/vendor.rs [hunk 1]
```diff
@@ -342,13 +342,7 @@ fn cp_sources(
...
-        // Join pathname components individually to make sure that the joined
-        // path uses the correct directory separators everywhere, since
-        // `relative` may use Unix-style and `dst` may require Windows-style
-        // backslashes.
-        let dst = relative
-            .iter()
-            .fold(dst.to_owned(), |acc, component| acc.join(&component));
+        let dst = dst.join(relative);
 
         paths::create_dir_all(dst.parent().unwrap())?;
```
Comment (epage) [inline src/cargo/ops/vendor.rs:347]: I assume this comment is relevant here

> I believe std::path::Path should be quite cross-platform compatible, as [it calls into something like [GetFullPathNameW](https://github.com/rust-lang/rust/blob/e760daa6a729b3d52a38804e9766f7d89dc27357/library/std/src/sys/path/windows.rs#L280-L287) eventually when being used. I might be wrong though.

I assume this was done for a reason though so I'm a bit cautious and would like input from others.

Comment (weihanglo) [reply src/cargo/ops/vendor.rs:347]: Seems like https://github.com/rust-lang/cargo/issues/6198 is relevant?

https://github.com/rust-lang/rust/blob/e760daa6a729b3d52a38804e9766f7d89dc27357/library/std/src/sys/path/windows.rs#L253

If it already contains `\\?\` then `GetFullPathNameW` won't be called. I'll drop the commit instead.

Comment (weihanglo) [reply src/cargo/ops/vendor.rs:347]: This evil behavior is deeply rooted in Cargo 😞.

**Pull request merged**